fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java

```diff
@@ -200,6 +200,11 @@ public void auditAfterExec(String origStmt, StatementBase parsedStmt, PQueryStat
                 .setStmtId(ctx.getStmtId())
                 .setIsForwardToLeader(isForwardToLeader)
                 .setQueryId(ctx.getQueryId() == null ? "NaN" : ctx.getQueryId().toString());
+        // Sometimes, error msg will be set in EOF state, such as query with limit,
+        // so here we only add error messages for query on ERROR status
+        if (ctx.getState().isError()) {
+            ctx.getAuditEventBuilder().setErrorMsg(ctx.getState().getErrorMessage());
```

Comment:
I would suggest strip it into a fixed-size string, just in case the error message become too long